isis/src/control/objs/BundleAdjust/BundleAdjust.cpp

```diff
@@ -2845,6 +2844,27 @@ namespace Isis {
   }
 
 
+  /**
+   * Return the updated model state for the ith cube in the cube list given to the
+   * constructor. This is only valid for CSM cubes.
+   *
+   * @param i The index of the cube to get the model state for
+   *
+   * @return @b QString The updated CSM model state string
+   */
+  QString BundleAdjust::modelState(int i) {
+    Camera *imageCam = m_controlNet->Camera(i);
+    if (imageCam->GetCameraType() != Camera::Csm) {
+      QString msg = "Cannot get model state for image [" + toString(i) +
+                    "] because it is not a CSM camera model.";
+      throw IException(IException::Programmer, msg, _FILEINFO_);
+    }
+
+    CSMCamera *csmCamera = dynamic_cast<CSMCamera*>(imageCam);
```

Comment:
I am not happy about having the dynamic_cast in BundleAdjust and not using the Observation, but attempts to use the polymorphism of the observation resulting in strange issues with the indices in the control network not matching the indices in the observations. I also ran into issues with who has the correct camera model and simultaneous open/close of the cube. This approach side-steps all of those issues at the cost of the cast happening here.